benchmark/scenarios/test/test_write_request_load.py

```diff
@@ -0,0 +1,362 @@
+        """
+        When the number of samples collected is less than the sample
+        size, the rate should be smaller than `req_per_second`.
+        """
+        r = WRateMeasurer()
+        req_per_second = 5
+
+        self.increase_rate(r, req_per_second, (r.sample_size / 2))
+
+        self.assertEqual((req_per_second * (r.sample_size / 2)) /
+                         r.sample_size,
+                         r.rate())
+
+    def test_rate_is_correct_when_enough_samples(self):
+        """
+        A WRateMeasurer should correctly report the rate when enough
+        samples have been collected.
+        """
+        r = WRateMeasurer()
+        req_per_second = 5
+
+        self.increase_rate(r, req_per_second, r.sample_size)
+
+        self.assertEqual(req_per_second, r.rate())
+
+    def test_old_samples_are_not_considered(self):
+        """
+        Old samples should be discarded, meaning that only `sample_size`
+        number of requests are considered for the rate, and when receiving
+        a new sample, the oldest one is discarded.
+        """
+        r = WRateMeasurer()
+        req_per_second = 5
+        # generate samples that should get lost
+        self.increase_rate(r, 100, r.sample_size/2)
+
+        # generate r.sample_size samples that will make the initial
+        # ones not count
+        self.increase_rate(r, req_per_second, r.sample_size)
+
+        self.assertEqual(req_per_second, r.rate())
+
+    def test_only_received_samples_considered_in_rate(self):
+        """
+        The rate should be based on the number of received requests,
+        not the number of sent requests.
+        """
+        r = WRateMeasurer()
+        send_per_second = 100
+        rec_per_second = 5
+
+        self.send_requests(r, send_per_second, r.sample_size)
+        self.receive_requests(r, rec_per_second, r.sample_size)
+
+        self.assertEqual(rec_per_second, r.rate())
+
+
+class RequestDroppingFakeFlockerClient(
+    proxyForInterface(IFlockerAPIV1Client)
+):
+    """
+    A FakeFlockerClient that can drop alternating requests.
+    """
+    def __init__(self, nodes):
+        super(RequestDroppingFakeFlockerClient, self).__init__(nodes)
+        self.drop_requests = False
+        self._dropped_last_request = False
+        self._real_nodes = nodes
+
+    def move_dataset(self, primary, dataset_id, configuration_tag=None):
+        if not self.drop_requests:
+            return super(RequestDroppingFakeFlockerClient, self).list_nodes()
+        else:
+            if self._dropped_last_request:
+                self._dropped_last_request = False
+                return super(RequestDroppingFakeFlockerClient,
+                             self).list_nodes()
+            self._dropped_last_request = True
+        return Deferred()
+
+class UnresponsiveDatasetCreationFakeFlockerClient(
+    proxyForInterface(IFlockerAPIV1Client)
+):
+    """
+    A FakeFlockerClient that can drop alternating requests.
```

Comment:
This description seems to be copied from `RequestDroppingFakeFlockerClient` while this class is doing something slightly different.